In [1]:
import pandas as pd

In [4]:
import os

print(os.getcwd())

C:\crawling_project


In [38]:
result = pd.read_csv('data/result.csv', encoding='cp949') 

# result.head(2)

In [12]:
result.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74678 entries, 0 to 74677
Data columns (total 27 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   입찰공고번호         74678 non-null  object 
 1   입찰공고차수         74678 non-null  int64  
 2   입찰공고명          74678 non-null  object 
 3   업무구분명          74678 non-null  object 
 4   계약체결형태명        74678 non-null  object 
 5   계약체결방법명        74678 non-null  object 
 6   낙찰자결정방법명       74678 non-null  object 
 7   적격심사여부         74678 non-null  object 
 8   공고기관명          74678 non-null  object 
 9   공고기관코드         74678 non-null  object 
 10  수요기관명          74678 non-null  object 
 11  수요기관코드         74678 non-null  object 
 12  낙찰하한율          63901 non-null  float64
 13  예정가격           51840 non-null  float64
 14  기초금액           73064 non-null  float64
 15  추정가격           74678 non-null  float64
 16  개찰일자           74678 non-null  object 
 17  개찰시각           74677 non-null  object 
 18  개찰결과구분

In [15]:
len(result.최종낙찰업체명.unique())

13757

##### 약 오만건의 결과중 최종낙찰업체 종류개수 13757 -> 한 업체 여러번 낙찰 가능성

#### 연결시 새 컬럼 생성(연도+공고번호)

In [52]:
연도공고번호 = []

for a, i in zip(result.개찰일자, result.입찰공고번호):
    b = a.split('-')
    c = b[0]
    연도공고번호.append(c + i)

result['연도공고번호'] = 연도공고번호



In [53]:
# result.head()

In [54]:
result_job = result[['연도공고번호','입찰공고명', '업무구분명', '낙찰자결정방법명', '계약체결방법명', '최종낙찰업체명','개찰일자','최종낙찰일자', '예정가격', '최종낙찰금액']].copy()
result_job.head()

연도공고번호                            입찰공고명 업무구분명 낙찰자결정방법명      계약체결방법명  \
0  2016MHD0102              상륙기동헬기 조종시뮬레이터 선행연구    용역       협상  협상에의한계약(서류)   
1  2016MHD0130               특수작전용 유탄발사기사업 선행연구    용역       협상  협상에의한계약(서류)   
2  2016UMM0003  00부대 물절약 투자대행사업(WASCO  15-A047)    용역       협상  협상에의한계약(전자)   
3  2016UMM0004  00부대 물절약 투자대행사업(WASCO  15-A048)    물품       협상  협상에의한계약(전자)   
4  2016UMM0860              16년 음향 영상장비 유지보수 용역    용역    적격심사제         일반경쟁   

          최종낙찰업체명        개찰일자      최종낙찰일자         예정가격       최종낙찰금액  
0  재단법인 한국군사문제연구원  2016-01-11  2016-02-11   30000000.0   28000000.0  
1    사단법인 안보경영연구원  2016-01-05  2016-01-06   74092000.0   71000000.0  
2    주식회사 도화엔지니어링  2016-01-29  2016-01-29  667000000.0  535467000.0  
3             NaN  2016-02-03         NaN          NaN          NaN  
4        주식회사 케이원  2016-01-05  2016-01-05  456378904.0  396923000.0

#### 예정가격과 최종낙찰금

In [55]:
result_job['낙찰금차'] = result_job['최종낙찰금액'] - result_job['예정가격']
result_job['하락비율'] = result_job['최종낙찰금액'] / result_job['예정가격']


In [56]:
result_job.head()

연도공고번호                            입찰공고명 업무구분명 낙찰자결정방법명      계약체결방법명  \
0  2016MHD0102              상륙기동헬기 조종시뮬레이터 선행연구    용역       협상  협상에의한계약(서류)   
1  2016MHD0130               특수작전용 유탄발사기사업 선행연구    용역       협상  협상에의한계약(서류)   
2  2016UMM0003  00부대 물절약 투자대행사업(WASCO  15-A047)    용역       협상  협상에의한계약(전자)   
3  2016UMM0004  00부대 물절약 투자대행사업(WASCO  15-A048)    물품       협상  협상에의한계약(전자)   
4  2016UMM0860              16년 음향 영상장비 유지보수 용역    용역    적격심사제         일반경쟁   

          최종낙찰업체명        개찰일자      최종낙찰일자         예정가격       최종낙찰금액  \
0  재단법인 한국군사문제연구원  2016-01-11  2016-02-11   30000000.0   28000000.0   
1    사단법인 안보경영연구원  2016-01-05  2016-01-06   74092000.0   71000000.0   
2    주식회사 도화엔지니어링  2016-01-29  2016-01-29  667000000.0  535467000.0   
3             NaN  2016-02-03         NaN          NaN          NaN   
4        주식회사 케이원  2016-01-05  2016-01-05  456378904.0  396923000.0   

          낙찰금차      하락비율  
0   -2000000.0  0.933333  
1   -3092000.0  0.958268  
2 -131533000.0  0.802799  
3          NaN       NaN  
4  -59455904.0  0.869722

In [51]:
up = (result_job['낙찰금차'] > 0).sum() # 예정가보다 최종낙찰금이 상승한경우(아주소수)
down = (result_job['낙찰금차'] < 0).sum() # 예정가보다 최종낙찰금이 하락한경우
print(up)
print(down)

29
51539


#### 물품 vs 용역 비교

In [61]:
# '업무구분명'별로 '최종낙찰금액'의 합계 구하기
result_job.groupby('업무구분명')['최종낙찰금액'].count() 

업무구분명
물품    42685
용역     9309
Name: 최종낙찰금액, dtype: int64

In [62]:
result_job.groupby('업무구분명')['최종낙찰금액'].sum() 

업무구분명
물품    2.056181e+13
용역    3.697169e+12
Name: 최종낙찰금액, dtype: float64

In [63]:
result_job.groupby('업무구분명')['최종낙찰금액'].mean() # 건당 최종낙찰금 물품이 더 높음

업무구분명
물품    4.817104e+08
용역    3.971607e+08
Name: 최종낙찰금액, dtype: float64

In [64]:
result_job.groupby('업무구분명')['하락비율'].mean() # 예상금액에서 최종낙찰금으로 하락한 정도는 둘다 비슷

업무구분명
물품    0.876652
용역    0.872330
Name: 하락비율, dtype: float64

#### 낙찰에 소요된 기간 

In [65]:
result_job['최종낙찰일자'] = pd.to_datetime(result_job['최종낙찰일자'])
result_job['개찰일자'] = pd.to_datetime(result_job['개찰일자'])


In [66]:
# 새 컬럼 생성 (개찰일 ~ 낙찰일 소요기간)
result_job['낙찰기간'] = result_job['최종낙찰일자'] - result_job['개찰일자']

In [73]:
result_job.낙찰기간.describe() # 평균 19시간 소요..? / 편차가 큼 -> 건별로 소요기간이 상이함(넓게퍼짐)


count                         51994
mean      0 days 19:31:19.555333307
std      11 days 06:45:07.113232398
min              -11 days +00:00:00
25%                 0 days 00:00:00
50%                 0 days 00:00:00
75%                 0 days 00:00:00
max              1629 days 00:00:00
Name: 낙찰기간, dtype: object

#### 낙찰 방법별 집계

In [74]:
result_job.낙찰자결정방법명.unique()

array(['협상', '적격심사제', '최저가격제'], dtype=object)

In [77]:
result_job.groupby('낙찰자결정방법명')['최종낙찰금액'].describe()

count          mean           std  min         25%         50%  \
낙찰자결정방법명                                                                     
적격심사제     43584.0  2.467994e+08  4.171196e+09  0.0  16882752.5  51649077.0   
최저가격제      6668.0  3.447837e+08  2.616320e+09  0.0   6871625.0  23016233.5   
협상         1742.0  6.431373e+09  4.851440e+10  0.0  40053500.0  79000000.0   

                   75%           max  
낙찰자결정방법명                              
적격심사제     1.419629e+08  7.050000e+11  
최저가격제     5.200000e+07  1.120000e+11  
협상        4.237500e+08  1.101950e+12